### ISSUE TITLE
[BUG] Shards cannot be recovered (stays unassigned) after rolling upgrade from OpenSearch 1.2.1 to 2.0.1

### ISSUE THREAD

**Issue by juntezhang** [opensearch-project/OpenSearch#3968]
**Describe the bug**
We updated the Docker image from OpenSearch 1.2.1 to OpenSearch 2.0.1. A rolling upgrade was performed where 1 node is shutdown per time and then upgraded, and then restarted. After upgrading all nodes to OpenSearch 2.0.1, we see that some shards cannot be recovered. They stay unassigned. 

After retrying to initalize the unassigned shards with the Reroute API (POST `_cluster/reroute?retry_failed`), the same shards stay unassigned.

After deleting the indices with unassigned shards, and restore these indices with the Snapshot API, they still stay unassigned.

An example error that I am getting is this:

```
{
  "state": "UNASSIGNED",
  "primary": true,
  "node": null,
  "relocating_node": null,
  "shard": 0,
  "index": "b640d9c7dcc44aa9a45198752b69ab75_standard.1654786318020",
  "recovery_source": {
    "type": "EXISTING_STORE",
    "bootstrap_new_history_uuid": false
  },
  "unassigned_info": {
    "reason": "MANUAL_ALLOCATION",
    "at": "2022-07-07T11:01:38.596Z",
    "delayed": false,
    "details": "failed shard on node [CSr1z8yvRxi8a56R-htrFw]: failed recovery, failure RecoveryFailedException[[b640d9c7dcc44aa9a45198752b69ab75_standard.1654786318020][0]: Recovery failed on {os-data-0}{CSr1z8yvRxi8a56R-htrFw}{rr1K1P64SPuaOC9VN5YtXA}{10.40.64.55}{10.40.64.55:9300}{dr}{zone=0, shard_indexing_pressure_enabled=true}]; nested: IndexShardRecoveryException[failed recovery]; nested: IllegalArgumentException[cannot change field \"Url___searchable.text\" from storeTermVector=false to inconsistent storeTermVector=true]; ",
    "allocation_status": "deciders_throttled"
  }
}
```

The text subfield is defined in the analyzer as follow:

```
  "text": {
    "analyzer": "icu_standard",
    "term_vector": "with_positions_offsets",
    "type": "text"
  }
```

**To Reproduce**
1. Index data on OpenSearch 1.2.1 into multple indices.
2. Do a rolling upgrade of the cluster to OpenSearch 2.0.1
3. Wait for recovery of shards.
4. See that some shards stay unassigned.
5. See error

**Expected behavior**
We expect that all existing shards can be initialized after doing a rolling upgrade.

**Plugins**
analysis-icu
analysis-kuromoji
analysis-nori
analysis-smartcn
analysis-stempel
analysis-ukrainian
repository-azure

**Screenshots**

**Host/Environment (please complete the following information):**
 - OS: Ubuntu (on pods in k8s)
 - Version latest

**Additional context**
This occurs not for all indices.

**Comment by andrross** [comment]
@juntezhang I believe this is an [issue that has been fixed in Lucene 9.2](https://issues.apache.org/jira/browse/LUCENE-10518). OpenSearch 2.1.0 has been released and uses Lucene 9.2. Can you try upgrading to OpenSearch 2.1 and see if that resolves your issue?

**Comment by juntezhang** [comment]
Thank you @andrross , I will check by updating to OS 2.1.0 and will report back. It looks like this is the issue indeed. 

> @juntezhang I believe this is an [issue that has been fixed in Lucene 9.2](https://issues.apache.org/jira/browse/LUCENE-10518). OpenSearch 2.1.0 has been released and uses Lucene 9.2. Can you try upgrading to OpenSearch 2.1 and see if that resolves your issue?

**Comment by juntezhang** [comment]
I believe this issue has been fixed in 2.1.0, so this bug can be closed.
